pkg/components/modelserver/modelserver.go

```diff
@@ -168,14 +176,29 @@ func NewService(k *v1alpha1.Kepler) *corev1.Service {
 			Selector:  labels,
 			Ports: []corev1.ServicePort{{
 				Name:       "http",
-				Port:       exporterPort,
+				Port:       int32(port),
 				TargetPort: intstr.FromString("http"),
 			}},
 		},
+	}, nil
+}
+
+func GetModelServerConfigForClient(k *v1alpha1.Kepler) map[string]string {
+	ms := k.Spec.ModelServer
+	if ms == nil {
+		return map[string]string{}
+	}
+
+	msConfig := k8s.StringMap{
+		"MODEL_SERVER_URL": defaultIfEmpty(ms.URL, GetDefaultModelServer(*k.Spec.ModelServer)),
 	}
+	msConfig = msConfig.AddIfNotEmpty("MODEL_SERVER_REQ_PATH", ms.RequestPath)
+	msConfig = msConfig.AddIfNotEmpty("MODEL_SERVER_MODEL_LIST_PATH", ms.ListPath)
+
+	return msConfig.ToMap()
 }
 
-func NewConfigMap(d components.Detail, k *v1alpha1.Kepler) *corev1.ConfigMap {
+func NewConfigMap(d components.Detail, k *v1alpha1.Kepler) (*corev1.ConfigMap, error) {
```

Comment:
```suggestion
func NewConfigMap(d components.Detail, ms v1alpha1.ModelServerSpec) (*corev1.ConfigMap) {
```

Returing error only helps to mask logical error (made by the developer) which is that this function should not  be called if modelserver is nil. By changing the arg to accept an object instead of pointer, we don't have to return error (keeping the signature of all New<X> the same).